src/source/src_hops.c

```diff
@@ -38,6 +38,20 @@
         
         obj->format = format_clone(src_hops_config->format);
         obj->interface = interface_clone(src_hops_config->interface);
+        if (src_hops_config->channel_map != NULL)
+        {
+            // Will not be null if in pulseaudio mode
+            memcpy(&obj->cm, src_hops_config->channel_map, sizeof(pa_channel_map));
+        }
+        else
+        {
+            // Can be null if not in pulseaudio mode ONLY
+            if (obj->interface->type == interface_pulseaudio)
+            {
+                printf("Error: Pulseaudio interface requires channel map.\n");
+                exit(EXIT_FAILURE);
+            }
```

Comment:
You can put the if next to the else. So, the code is less indented.

Reply:
Good point